Source/Extensions/Blazorise.DataGrid/DataGrid.razor.cs

```diff
@@ -947,17 +947,26 @@ protected IEnumerable<DataGridColumn<TItem>> DisplayableColumns
         /// <summary>
         /// Returns true if EmptyTemplate is set and Data is null or empty.
         /// </summary>
-        protected bool IsEmptyTemplateVisible => !IsLoadingTemplateVisible && !IsNewItemInGrid && EmptyTemplate != null && ( Data == null || !Data.Any() );
+        protected bool IsEmptyTemplateVisible
+            => !IsLoadingTemplateVisible && !IsNewItemInGrid && EmptyTemplate != null && ( Data == null || !Data.Any() );
+
+        /// <summary>
+        /// Returns true if EmptyFilterTemplate is set and FilteredData is null or empty.
+        /// </summary>
+        protected bool IsEmptyFilterTemplateVisible
+            => !IsLoadingTemplateVisible && !IsNewItemInGrid && EmptyFilterTemplate != null && ( !( Data == null || !Data.Any() ) && ( FilteredData == null || !FilteredData.Any() ) );
 
         /// <summary>
         /// Returns true if ShowPager is true and grid is not empty or loading.
         /// </summary>
-        protected bool IsPagerVisible => ShowPager && !IsLoadingTemplateVisible && ( ( IsButtonRowVisible && ButtonRowTemplate != null ) || !IsEmptyTemplateVisible );
+        protected bool IsPagerVisible
+            => ShowPager && !IsLoadingTemplateVisible && ( ( IsButtonRowVisible && ButtonRowTemplate != null ) || !IsEmptyTemplateVisible );

```

Comment:
I have also been doing this approach with property lambdas, looks and reads way better!

Reply:
I agree, you can notice right away that is a lambda property.